simplified-viewer-audiobook/src/main/java/org/nypl/simplified/viewer/audiobook/AudioBookPlayerActivity.kt

```diff
@@ -336,7 +351,7 @@ class AudioBookPlayerActivity :
         uri = null
       )
 
-      if (event.isLocalBookmark) {
+      if (event is PlayerEventCreateBookmark && event.isLocalBookmark) {
```

Comment:
I think this approach may be a bit tricky because we are accepting an event of the type `PlayerEvent.PlayerEventWithSpineElement` which is basically the super type of the events. Which means this method would send a remote bookmark to the server on every event that is different from the `PlayerEventCreateBookmark` and we would have to get an eye on it every time we changed the `onPlayerEvent` method. 

My suggestion is:
- we keep the `savePlayerPosition(event: PlayerEventCreateBookmark)` as the only method and with no changes;
- on the `onPlayerEvent(event: PlayerEvent)` method we create a PlayerEventCreateBookmark event for every new handled event you added (`PlayerEventPlaybackStarted`, `PlayerEventPlaybackPaused`, `PlayerEventPlaybackStopped`) with the `isLocalBookmark` as false;

What's your opinion on that?

Reply:
Oh! That makes sense to me. I'll make that change and follow up!